modules/apps/27-interchain-accounts/host/keeper/relay.go

```diff
@@ -25,17 +25,16 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet) ([]byt
 	case icatypes.EXECUTE_TX:
 		msgs, err := icatypes.DeserializeCosmosTx(k.cdc, data.Data)
 		if err != nil {
-			return nil, err
+			return nil, sdkerrors.Wrapf(err, "cannot deserialize cosmos Tx")
 		}
 
 		txResponse, err := k.executeTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs)
 		if err != nil {
-			return nil, err
+			return nil, sdkerrors.Wrapf(err, "error executing Tx")
 		}
-
 		return txResponse, nil
 	default:
-		return nil, icatypes.ErrUnknownDataType
+		return nil, sdkerrors.Wrapf(icatypes.ErrUnknownDataType, "unknown data type")
```

Comment:
This looks redundant to me, I believe the error output will be `failed to execute message 0: unknown data type: unknown data type" because [`ErrUnknownDataType`](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/types/errors.go#L8) already contains this message. My general rule of thumb is to only wrap errors if you are adding information to the error that occurred. We should expect the errors returned to have some base information on what went wrong